lib/rules/class-methods-use-this.js

```diff
@@ -125,10 +157,23 @@ module.exports = {
             "FunctionDeclaration:exit": exitFunction,
             FunctionExpression: enterFunction,
             "FunctionExpression:exit": exitFunction,
-            "PropertyDefinition > ArrowFunctionExpression.value": enterFunction,
-            "PropertyDefinition > ArrowFunctionExpression.value:exit": exitFunction,
+
+            /*
+             * Class field value are implicit functions.
+             */
+            PropertyDefinition: pushContext,
+            "PropertyDefinition:exit": popContext,
+            "PropertyDefinition > *.key": popContext,
+            "PropertyDefinition > *.key:exit": pushContext,
```

Comment:
```suggestion
            "PropertyDefinition > *.key:exit": pushContext,
            "PropertyDefinition:exit": popContext,
```

`PropertyDefinition: pushContext` - `"PropertyDefinition > *.key": popContext` pair seems unnecessary?